cpg_workflows/jobs/align.py

```diff
@@ -262,13 +265,20 @@ def align(
         requested_nthreads=requested_nthreads,
         markdup_tool=markdup_tool,
         output_path=output_path,
+        sorted_bam_path=sorted_bam_path,
         out_markdup_metrics_path=out_markdup_metrics_path,
         align_cmd_out_fmt=output_fmt,
         overwrite=overwrite,
     )
     if md_j and md_j != merge_or_align_j:
         jobs.append(md_j)
 
+    if md_j:
+        if md_j.attributes.get('reusing_sorted_bam'):
+            # Remove all jobs except markdup if sorted bam is reused
+            logging.info(f'Reusing sorted bam from temp - only markdup job will be submitted for {sequencing_group.id}')
+            jobs = [md_j]
```

Comment:
II think this is fine, but I'd be keen to leave this as a logging statement based on the md_j attributes, then flow down to the existing `return jobs` below. If the task is in a position to resume then we should never have more than one job to return, so this is fine, but I'm scared of the way that this structural change (only return one job, which _should_ be 1 of 1) could interfere with workflow scheduling (inter-stage dependencies are only set based on the jobs returned by the Stage, not the all scheduled jobs in Hail)